default/python/AI/ResourcesAI.py

```diff
@@ -52,23 +59,64 @@ def _focus_name(focus: str) -> str:
     return _known_names.get(focus, focus)
 
 
+class Output(NamedTuple):
+    industry: float
+    research: float
+    influence: float
+    stability: float
+    value: float
+
+
 class PlanetFocusInfo:
     """The current, possible and future foci and output of one planet."""
 
-    def __init__(self, planet):
+    def __init__(self, planet: fo.planet):
         self.planet = planet
         self.current_focus = planet.focus
-        self.current_output = (
-            planet.currentMeterValue(fo.meterType.industry),
-            planet.currentMeterValue(fo.meterType.research),
-            planet.currentMeterValue(fo.meterType.influence),
+        self.current_output = Output(
+            industry=planet.currentMeterValue(fo.meterType.industry),
+            research=planet.currentMeterValue(fo.meterType.research),
+            influence=planet.currentMeterValue(fo.meterType.influence),
+            stability=planet.currentMeterValue(fo.meterType.happiness),
+            value=0.0,  # cannot calculate the value here, do we even need it?
         )
         self.possible_output = {}
-        industry_target = planet.currentMeterValue(fo.meterType.targetIndustry)
-        research_target = planet.currentMeterValue(fo.meterType.targetResearch)
-        influence_target = planet.currentMeterValue(fo.meterType.targetInfluence)
-        self.possible_output[self.current_focus] = (industry_target, research_target, influence_target)
         self.future_focus = self.current_focus
+        self.best_output_focus = None
+        self.second_best_focus = None
+        self.options = {focus for focus in planet.availableFoci if focus in supported_foci}
+
+    def __str__(self):
+        return str(self.planet)
+
+    def __repr__(self):
+        return self.planet.__repr__()
```

Comment:
Let's try to stick to convention: https://docs.python.org/3/reference/datamodel.html#object.__repr__

```suggestion
        return f"PlanetFocusInfo({repr(self.planet)})"
```

Minor notice:  in Python, you should not call magic methods directly, except the inheritance purposes.  usually There are another way to call them 
-  `__repr__` - > `repr()`
-  `__len__` -> `len()`
-  `__add__` -> `+`